### ISSUE TITLE
Lint/NestedMethodDefinition shouldn't flag singleton methods

### ISSUE THREAD

**Issue by CyborgMaster** [bbatsov/rubocop#2898]
I have the following method that dynamically creates a subclass
``` ruby
  def setup
    subclass = Class.new SimpleCommunicationSetting do
      def self.template
        {
          name: 'test',
        }
      end
    end
  end
```

which gets flagged with the following offense:
```
test/models/comm_engine/communication_settings/simple_communication_setting_test.rb:6:7: W: Lint/NestedMethodDefinition: Method definitions must not be nested. Use lambda instead.  (https://github.com/bbatsov/ruby-style-guide#no-nested-methods)
      def self.template
      ^^^^^^^^^^^^^^^^^
```

It appears that there should be a test for this: `does not register offense for nested definition inside Class.new`, but maybe the passed in superclass is confusing it?

RuboCop version
```
$ rubocop -V
0.37.2 (using Parser 2.3.0.5, running on ruby 2.2.2 x86_64-darwin14)
```

**Comment by segiddins** [comment]
I've got a fix for this I'll PR in a second

**Comment by CyborgMaster** [comment]
Thanks! You guys rock!

**Comment by CyborgMaster** [comment]
Sweet.  May we please have a bugfix release to RubyGems?
